### ISSUE TITLE
How to check if page is readable?

### ISSUE THREAD

**Issue by Feelnoobskill** [luin/readability#78]
I want to check if the page is readable or not. Is that possible?

**Comment by haroldtreen** [comment]
What do you mean by this? 

When you use readability, the article returned will be null if nothing was found. Otherwise it will return whatever part of the article it determined to be the main content.

I've been wondering how to flag articles that aren't being extracted properly (eg. a description has been extracted rather then the article). My current approach is to look at the size of the article html vs. input html. I've found that if the content html is < 3% of the original article - chances are the main article was missed.

Does that help @Feelnoobskill ? Or maybe you can elaborate what you picture the solution looking like?

**Comment by Feelnoobskill** [comment]
@haroldtreen  thanks for the replay. Basically, I would like to create reader mode like  [iOS Safari](https://cnet1.cbsistatic.com/img/Ntij3vj0Q49YcnkyELI5GRFSJJE=/2015/09/03/a35b1c5e-0c3e-4ae0-b3e2-6a016aa3b6f3/safari-reader-colors-ios-9-2.jpg) has. 

Meaning that some pages are not suitable for opening in reader mode (for example stackoverflow home page). Right now `node-readability` will extract some random text from webpage and this is not acceptable in my case . So i was thinking maybe someone already faced with this problem and can share their experience.

**Comment by haroldtreen** [comment]
Ah. Interesting. I wasn't aware that iOS did that. 

Some ideas: 
- You could look at the metadata to determine if the page is an article. Use that to remove the reader/force-show the reader button. For example:
```html
<meta property="og:type" content="article">
```
- You could run readability and check the % reduction. As stated before, I find ~2.5-3% to be a good metric for something going wrong.
- Run readability and look at the length of the output. Reader mode will probably only be useful if the length the output is > X characters/lines.

The tag stuff might be the closest stuff to being able to say yes/no without actually running the algorithms on the page.

**Comment by NinoSkopac** [comment]
This is good info @haroldtreen. 

It would be great if the library had an API for this (eg `isReadable`)
